src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -786,3 +787,67 @@ export const triggerVirusScanning = (
     return errAsync(new VirusScanFailedError())
   })
 }
+
+/**
+ * Downloads file from clean bucket
+ * @param cleanFileKey object key of the file in the clean bucket
+ * @param versionId id for versioning of the file in the clean bucket
+ * @returns okAsync(buffer) if file has been successfully downloaded from the clean bucket
+ * @returns errAsync(DownloadCleanFileFailedError) if file download failed
+ */
+export const downloadCleanFile = (cleanFileKey: string, versionId: string) => {
+  const logMeta = {
+    action: 'downloadCleanFile',
+    cleanFileKey,
+    versionId,
+  }
+
+  if (!validate(cleanFileKey)) {
+    logger.error({
+      message: 'Invalid clean file key - not a valid uuid',
+      meta: logMeta,
+    })
+
+    return errAsync(new InvalidFileKeyError())
+  }
+
+  let buffer = Buffer.alloc(0)
+
+  const writeStream = new Writable({
+    write(chunk, encoding, callback) {
+      buffer = Buffer.concat([buffer, chunk])
+      callback()
+    },
+  })
+
+  const readStream = AwsConfig.s3
+    .getObject({
+      Bucket: AwsConfig.virusScannerCleanS3Bucket,
+      Key: cleanFileKey,
+      VersionId: versionId,
+    })
+    .createReadStream()
```

Comment:
clarification: what was the consideration behind having a read stream pipe to the write stream, and then resolving the promise on stream end; might it be simpler to do something like:

```
let Body
try {
  const { Body } = await AwsConfig.s3
    .getObject({
      Bucket: AwsConfig.virusScannerCleanS3Bucket,
      Key: cleanFileKey,
      VersionId: versionId,
    })
    .promise()
} catch (e) {
    ... //Handle error here
}
```
were there any specific advantages you were trying to achieve?

Reply:
I was thinking about memory, but in this case, I think using streams won't make a difference oop 😅 will refactor to use promise!

Reply:
Oh I realised why I went down this path in the first place, because `S3.GetObjectOutput.Body` can be `Buffer|Uint8Array|Blob|string|Readable`, and it'll all be `Buffer` if I just use the streams 😅 + `response.content` accepts `Buffer` type.

Reply:
There seems to be no easier way to handle all the different types, sadly... https://stackoverflow.com/questions/25812384/how-do-determine-the-type-of-data-in-s3-getobject

Reply:
Ok, can go with this approach then